packages/locale/src/format/cldr/CldrCache.js

```diff
@@ -0,0 +1,54 @@
+/*
+* @copyright (c) 2020, Philipp Thuerwaechter & Pattrick Hueper
+* @license BSD-3-Clause (see LICENSE.md in the root directory of this source tree)
+*/
+
+import cldrData from 'cldr-data';
+import Cldr from 'cldrjs';
+
+const cldrDataLoaded = new Set();
+/**
+* Loads the Cldr data for the given path if it hasn't been loaded before.
+*/
+export const loadCldrData = (path) => {
+    if (!cldrDataLoaded.has(path)) {
+        Cldr.load(cldrData(path));
+        cldrDataLoaded.add(path);
+    }
+};
+
+const localeToCldrInstanceCache = {};
+/**
+* Returns a Cldr instance for the given locale.
+* Memoized, so a given locale will always return the exact same cldr instance.
+*/
+export const getOrCreateCldrInstance = (locale) => {
+    if (localeToCldrInstanceCache[locale] == null) {
+        localeToCldrInstanceCache[locale] = new Cldr(locale);
+    }
+    return localeToCldrInstanceCache[locale];
+};
+
+const cldrInstanceToMapZonesCache = new Map();
```

Comment:
This cache I imagine could be a bit bigger, since it's making a new `mapZones` object for each provided cldr instance. I'm not sure how often this even ends up getting used when printing a timezone? So I could probably just get rid of this cache and most cases wouldn't see a major performance difference.

Reply:
Each new `mapZones` objects appears to be about 20KB.  I'm not quite sure on the context of this code - is it likely that we generate different `mapZones` objects for many different locales over the lifetime of a program? Or is it typically only a few locales? If the number is small enough, probably fine to keep this cache around

![image](https://user-images.githubusercontent.com/1400151/88469125-ae7f3580-cebb-11ea-849c-2d9ae0f8ec5d.png)

Reply:
Since the normal use case should be only handfull of locales, it shouldn't be a problem. And even with hundreds of locales its just some MB, shouldn't harm a modern browser.

LGTM 👍